modules/claims_api/app/models/claims_api/power_of_attorney.rb

```diff
@@ -18,6 +18,17 @@ class PowerOfAttorney < ApplicationRecord
     before_validation :set_md5
     validates :md5, uniqueness: true
 
+    def self.find_using_identifier_and_source(source_name:, id: nil, header_md5: nil, md5: nil)
+      primary_identifier = {}
+      primary_identifier[:id] = id if id.present?
+      primary_identifier[:header_md5] = header_md5 if header_md5.present?
+      primary_identifier[:md5] = md5 if md5.present?
+      poa = ClaimsApi::PowerOfAttorney.find_by(primary_identifier)
+      return nil if poa.present? && poa.source_data['name'] != source_name
```

Comment:
Why not actually search by source since you added the index specifically for including it in the find_by? Is there a particular strategy here? Including the source rules out duplicate md5's and other potential markers and I'm not saying this is a bad option to doing it this way, just want to understand the logic for not including source_name in the search itself?

Reply:
I added the index to AutoEstablishedClaim which had source available as a column so was easy to just add to the find_by.
PowerOfAttorney stores the source name a bit differently. It has a column named "source_data" that is encrypted within the database. I spent a little time on it but couldn't get any sort of direct query to work but im definitely open to any ideas improving this logic, this is the best i could come up with to mimic the find_by logic youd expect.

Reply:
Makes logical sense.